File: urbanairship-test/src/main/java/com/urbanairship/TestPushProviders.java
package com.urbanairship;

import com.urbanairship.push.PushProvider;

import java.util.Collections;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class TestPushProviders extends PushProviders {

    public TestPushProviders(@NonNull AirshipConfigOptions config) {
        super(config);
    }

    @Nullable
    @Override
    public PushProvider getProvider(int platform, @NonNull String providerClass) {
        return null;
    }

    @NonNull
    @Override
    public List<PushProvider> getAvailableProviders() {
        return Collections.emptyList();
    }

    @Nullable
    @Override
    public PushProvider getBestProvider(int platform) {
        return null;
    }

    @Nullable
    @Override
    PushProvider getBestProvider() {
        return null;
    }

}

File: urbanairship-test/src/main/java/com/urbanairship/TestPushProviders.kt
package com.urbanairship

import com.urbanairship.push.PushProvider

public class TestPushProviders public constructor(config: AirshipConfigOptions) :
    PushProviders(config) {

    override fun getProvider(platform: Platform, providerClass: String): PushProvider? {
        return null
    }

    override fun getAvailableProviders(): List<PushProvider> {
        return emptyList()
    }

    override fun getBestProvider(platform: Platform): PushProvider? {
        return null
    }

    override val bestProvider: PushProvider? = null
}
